fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/service/serialization/serializer/loan/LoanDelinquencyRangeChangeBusinessEventSerializer.java

```diff
@@ -104,7 +116,79 @@ public <T> ByteBufferSerializable toAvroDTO(BusinessEvent<T> rawEvent) {
                 .setAmount(amount)//
                 .setCurrency(currencyMapper.map(data.getCurrency()))//
                 .setDelinquentDate(delinquentDate)//
-                .build();
+                .setInstallmentDelinquencyBuckets(installmentsDelinquencyData).build();
```

Comment:
This should be calculated and set only when the installment level delinquency is enabled on the loan.

https://github.com/apache/fineract/blob/f089f011168e09abeedc53313bde2b24c4239da2/fineract-loan/src/main/java/org/apache/fineract/portfolio/loanproduct/domain/LoanProduct.java#L213

Reply:
Ah, please ignore, I can see that you are checking it in line 124

Reply:
@reluxa , This check is there in calculateInstallmentLevelDelinquencyData method.

 private List<LoanInstallmentDelinquencyBucketDataV1> calculateInstallmentLevelDelinquencyData(Loan loan, LoanAccountData data) {
        List<LoanInstallmentDelinquencyBucketDataV1> loanInstallmentDelinquencyData = new ArrayList<>();
        if (loan.isEnableInstallmentLevelDelinquency()) {

Reply:
okay